rust/datafusion/src/optimizer/constant_folding.rs

```diff
@@ -627,7 +554,7 @@ mod tests {
 
         let expected = "\
         Projection: #a\
-        \n  Filter: NOT NOT #b\
+        \n  Filter: #b\
```

Comment:
This seems like an improvement to me `NOT NOT #b` is the same as `b` :) I suspect something was not quite right with the recursion previously

Reply:
Yeah, my original implementation did the tree traversal wrong for not expr :P It was doing a preorder traversal, which requires a convergent loop to produce `#b` in this case. Nice catch.